crates/re_ui/src/list_item2/property_content.rs

```diff
@@ -176,35 +175,67 @@ impl ListItemContent for PropertyContent<'_> {
-        );
-
-        let content_width =
-            (context.rect.width() - action_button_rect.width() - ReUi::text_to_icon_padding())
-                .at_least(0.0);
-
-        //TODO(ab): adaptable columns
-        let column_width = ((content_width - ReUi::text_to_icon_padding()) / 2.).at_least(0.0);
+        // │                                                                              │
+        // │◀─────────────────────────────background_x_range─────────────────────────────▶│
+        // │                                                                              │
+        // │ ◀───────────state.left_column_width────────────▶│┌──COLUMN_SPACING           │
+        // │                                                  ▼                           │
+        // │                       ◀──────────────CONTENT────┼──────────────────────────▶ │
+        // │ ┌ ─ ─ ─ ─ ┬ ─ ─ ─ ─ ┬ ┬────────┬─┬─────────────┬─┬─────────────┬─┬─────────┐ │
+        // │                       │        │ │             │││             │ │         │ │
+        // │ │         │         │ │        │ │             │ │             │ │         │ │
+        // │   INDENT       ▼      │  ICON  │ │    LABEL    │││    VALUE    │ │   BTN   │ │
+        // │ │         │         │ │        │ │             │ │             │ │         │ │
+        // │                       │        │ │             │││             │ │         │ │
+        // │ └ ─ ─ ─ ─ ┴ ─ ─ ─ ─ ┴ ┴────────┴─┴─────────────┴─┴─────────────┴─┴─────────┘ │
+        // │ ▲                     ▲         ▲               │               ▲            │
+        // │ └──state.left_x       │         └───────────────────────────────┤            │
+        // │                       │                         ▲               │            │
+        // │       content_left_x──┘           mid_point_x───┘     text_to_icon_padding   │
+        // │                                                                              │
+        //
+        // content_indent = content_left_x - state.left_x
+        // left_column_width = content_indent + icon_extra + label_width + COLUMN_SPACING/2
+
+        let state = super::StateStack::top(ui.ctx());
+
+        let content_left_x = context.rect.left();
+        // Total indent left of the content rect. This is part of the left column width.
+        let content_indent = content_left_x - state.left_x;
+        let mid_point_x = state.left_x
+            + state
+                .left_column_width
+                .unwrap_or_else(|| content_indent + (context.rect.width() / 2.).at_least(0.0));
+
+        let icon_extra = if icon_fn.is_some() {
+            ReUi::small_icon_size().x + ReUi::text_to_icon_padding()
+        } else {
+            0.0
+        };
 
         let icon_rect = egui::Rect::from_center_size(
             context.rect.left_center() + egui::vec2(ReUi::small_icon_size().x / 2., 0.0),
             ReUi::small_icon_size(),
         );
```

Comment:
no icon, no icon_rect

```suggestion
        let icon_rect = icon_fn.is_some().then(|| egui::Rect::from_center_size(
            context.rect.left_center() + egui::vec2(ReUi::small_icon_size().x / 2., 0.0),
            ReUi::small_icon_size(),
        ));
```

Reply:
in fact, better yet: don't construct it until it i s needed (inside of the `if let Some(icon_fn) = icon_fn {`).
It's always a good idea to reduce the scope of bindings, _and_ to reduce the amount of bindings in each scope